docs/ternary-operator.md

```diff
@@ -0,0 +1,165 @@
+# Ternary (Conditional) Operator: ?
+The ternary (conditional) operator is the only BrighterScript operator that takes three operands: a condition followed by a question mark (?), then an expression to execute (consequent) if the condition is true followed by a colon (:), and finally the expression to execute (alternate) if the condition is false. This operator is frequently used as a shortcut for the if statement. It can be used in assignments, and in any other place where an expression is valid. Due to ambiguity in the brightscript syntax, ternary operators cannot be used as standalone statements. See the [No standalone statements](#no-standalone-statements) for more information.
+
+## Basic usage
+
+```BrighterScript
+a = user = invalid ? "no user" : "logged in"
+```
+
+transpiles to:
```

Comment:
I think these explanations are a bit confusing.

What about going with:
1. ternary operator with literal values: transpiles with `bslib_ternary`
2. ternary operator with references/function calls: transpiles into a self-executing function with scope capture

Reply:
Which parts are "thought process" like? I wanted to be transparent in the way the transpiler works, and why we chose the implementations that we did. If there are specific sections that are too verbose, point them out and I'll see if there's a way to trim them down. but I did a quick read-through and still thought most of the information was generally relevant to someone reading the docs.